Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -3582,9 +3582,7 @@ private async Task<PipExecutionStep> ExecutePipStep(RunnablePip runnablePip)
                         {
                             var perfInfo = executionResult.PerformanceInformation;
 
-                            if (perfInfo != null)

```

Comment:
>perfInfo [](start = 32, length = 8)

I guess this is guarded against being null by virtue of EnsureSealed() in the getter. Worth leaving an assert in there for the comfort of future readers?

Reply:
we already check whether this is null just above: 

 (executionResult.PerformanceInformation != null)

---
In reply to: [339821083](https://github.com/microsoft/BuildXL/pull/1113#discussion_r339821083) [](ancestors = 339821083)